met-web/src/components/engagement/view/widgets/Timeline/TimelineWidgetView.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import { MetPaper, MetHeader2, MetParagraph, MetHeader4 } from 'components/common';
+import { Grid, Skeleton, Divider } from '@mui/material';
+import { Widget } from 'models/widget';
+import { useAppDispatch } from 'hooks';
+import { openNotification } from 'services/notificationService/notificationSlice';
+import { TimelineWidget, TimelineEvent } from 'models/timelineWidget';
+import { fetchTimelineWidgets } from 'services/widgetService/TimelineService';
+
+interface TimelineWidgetProps {
+    widget: Widget;
+}
+
+const TimelineWidgetView = ({ widget }: TimelineWidgetProps) => {
+    const dispatch = useAppDispatch();
+    const [timelineWidget, setTimelineWidget] = useState<TimelineWidget>({
+        id: 0,
+        widget_id: 0,
+        engagement_id: 0,
+        title: '',
+        description: '',
+        events: [],
+    });
+    const [isLoading, setIsLoading] = useState(true);
+
+    const fetchTimeline = async () => {
+        try {
+            const timelines = await fetchTimelineWidgets(widget.id);
+            const timeline = timelines[timelines.length - 1];
+            setTimelineWidget(timeline);
+            setIsLoading(false);
+        } catch (error) {
+            setIsLoading(false);
+            console.log(error);
+            dispatch(
+                openNotification({
+                    severity: 'error',
+                    text: 'Error occurred while fetching Engagement widgets information',
+                }),
+            );
+        }
+    };
+
+    useEffect(() => {
+        fetchTimeline();
+    }, [widget]);
+
+    const handleRenderTimelineEvent = (tEvent: TimelineEvent, index: number) => {
+        const containerStylesObject = {
+            minHeight: index + 1 == timelineWidget.events.length ? '60px' : '80px',
```

Comment:
```suggestion
            minHeight: index + 1 === timelineWidget.events.length ? '60px' : '80px',
```

These comparisons should be using strict equality